torchrl/data/tensordict/tensordict.py

```diff
@@ -1921,17 +1904,8 @@ def batch_dims(self, value: COMPATIBLE_TYPES) -> None:
         )
 
     @property
-    def device(self) -> torch.device:
-        device = self._device
-        if device is None:
-            raise RuntimeError(
-                "Querying a tensordict device when it has not been set is not "
-                "permitted. Use `tensordict.device_safe()` return the device "
-                "value or `None` if the device is not set.\n\nSet the device "
-                "upon creation using the `device=dest` keyword, or "
-                "subsequently using `tensordict.to(device)."
-            )
-        return device
+    def device(self) -> Union[None, torch.device]:
```

Comment:
Could you add a note here in the docstring, mentioning that we allow device to be None if it is not specified upon construction or when calling `tensordict.to(...)`?

Reply:
Added the comment:
        """Returns `None` if device hasn't been provided in the constructor
        or set via `tensordict.to(device)`.
        """